src/IRrecv.cpp

```diff
@@ -423,20 +417,14 @@ void IRrecv::disableIRIn(void) {
 #ifndef UNIT_TEST
 #if defined(ESP8266)
   os_timer_disarm(&timer);
-#endif  // ESP8266
-#if defined(ESP32)
-  // Check for ESP32 core version and handle timer functions differently
-  #if ARDUINO_COREV3
-    // For ESP32 core version 3.x
-    timerWrite(timer, 0);  // Reset the timer
-    timerDetachInterrupt(timer);  // Detach the interrupt
-    timerEnd(timer);  // End the timer
-  #else
-    // For ESP32 core version 2.x
-    timerAlarmDisable(timer);  // Disable the alarm
-    timerDetachInterrupt(timer);  // Detach the interrupt
-    timerEnd(timer);  // End the timer
-  #endif
+#elif defined(_ESP32_ARDUINOV3)
+  timerWrite(timer, 0);  // Reset the timer
+  timerDetachInterrupt(timer);
+  timerEnd(timer);
+#elif defined(ESP32)
+  timerAlarmDisable(timer);
+  timerDetachInterrupt(timer);
+  timerEnd(timer);
 #endif  // ESP32
```

Comment:
Suggestion: Try to avoid `#elsif` where possible
```c++
#endif  // ESP8266
#if defined(ESP32)
  // Check for ESP32 core version and handle timer functions differently
#if defined(_ESP32_ARDUINOV3)
  // Code for Arduino cores V3 and beyond
  timerWrite(timer, 0);  // Reset the timer
  timerDetachInterrupt(timer);
  timerEnd(timer);
#else  // _ESP32_ARDUINOV3
  // Code for Arduino cores before V3
  timerAlarmDisable(timer);
  timerDetachInterrupt(timer);
  timerEnd(timer);
#endif  // _ESP32_ARDUINOV3
#endif  // ESP32
```

Reply:
Any particular reason to why? it is becoming messy quite fast, and it is used elsewhere already.
I am however considering that we should split out this logic to a separate file or something to reduce the mess a bit.